src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+  // loop over all volumes
+  for (int i = 1; i <= num_vols; ++i) {
+    int volid = DAG->id_by_index(3, i);
+    moab::EntityHandle eh = DAG->entity_by_index(3, i);
+
+    // vector of tally values
+    tally_assignment = tally_assignments[eh];
+    // set the value of each string for
+    std::string tally = "|";
+    for (int j = 0 ; j < tally_assignment.size() ; j++) {
+      // delimit each particle/value pair with a pipe symbol
+      tally += tally_assignment[j] + "|";
+    }
+    tally_data_eh[eh] = tally;
+  }
+}
+
+// parse the boundary data
+void dagmcMetaData::parse_boundary_data() {
+  std::map<moab::EntityHandle, std::vector<std::string> > boundary_assignments;
+  boundary_assignments = get_property_assignments("boundary", 2, ":");
+
+  int num_surfs = DAG->num_entities(2);
+
+  std::vector<std::string> boundary_assignment; // boundary conditions for the current entity
+  // loop over all surfaces
+  for (int i = 1; i <= num_surfs; ++i) {
+    int surfid = DAG->id_by_index(2, i);
+    moab::EntityHandle eh = DAG->entity_by_index(2, i);
+
+    boundary_assignment = boundary_assignments[eh];
+    if (boundary_assignment.size() != 1) {
+      std::cout << "More than one boundary conditions specified for " << surfid << std::endl;
+      std::cout << surfid << " has the following density assignments" << std::endl;
+      for (int j = 0 ; j < boundary_assignment.size() ; j++) {
+        std::cout << boundary_assignment[j] << std::endl;
+      }
+      std::cout << "Please check your boundary condition assignments " << surfid << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    // 2d entities have been tagged with the boundary condition property
+    // ie. both surfaces and its members triangles,
+
+    if (boundary_assignment[0].find("Reflecting") != std::string::npos)
+      surface_boundary_data_eh[eh] = "Reflecting";
+    if (boundary_assignment[0].find("White") != std::string::npos)
+      surface_boundary_data_eh[eh] = "White";
+  }
+}
+
+// parse the surface tally data from the file
+void dagmcMetaData::parse_tally_surface_data() {
+  std::map<moab::EntityHandle, std::vector<std::string> > tally_assignments;
+  tally_assignments = get_property_assignments("tally", 2, ":");
+
+  int num_surfaces = DAG->num_entities(2);
+
+  std::vector<std::string> tally_assignment; // surface tally assignments for the current entity
+  // loop over all volumes
+  for (int i = 1; i <= num_surfaces; ++i) {
+    int surfid = DAG->id_by_index(2, i);
+    moab::EntityHandle eh = DAG->entity_by_index(2, i);
+
+    // vector of tally values
+    tally_assignment = tally_assignments[eh];
+    // set the value of each string for
+    std::string tally = "|";
+    for (int j = 0 ; j < tally_assignment.size() ; j++) {
+      // delimit each particle/value pair with a pipe symbol
+      tally += tally_assignment[j] + "|";
+    }
+    tally_data_eh[eh] = tally;
+  }
+}
+
+
+// for a given property with a range of delimiters, and dimensionality
```

Comment:
Potential incomplete comment: Did you mean `, get property assignments and dimensionality` rather than `, and dimensionality`?